nifi-nar-bundles/nifi-redis-bundle/nifi-redis-extensions/src/test/java/org/apache/nifi/redis/service/ITRedisDistributedMapCacheClientService.java

```diff
@@ -64,22 +73,154 @@
+    }
+
+    @Test
+    public void testSentinelRedis() throws InitializationException, IOException {
+        RedisContainer redisMasterContainer = setupStandaloneRedis(null);
+        String masterHost = "127.0.0.1";
+        int masterPort = redisMasterContainer.port;
+        setUpRedisReplica(masterHost, masterPort, null);
+        setUpRedisReplica(masterHost, masterPort, null);
+
+        int sentinelAPort = setUpSentinel(masterHost, masterPort, null, 2, null).port;
+        int sentinelBPort = setUpSentinel(masterHost, masterPort, null, 2, null).port;
+        int sentinelCPort = setUpSentinel(masterHost, masterPort, null, 2, null).port;
+
+        setUpRedisConnectionPool(portsToConnectionString(sentinelAPort, sentinelBPort, sentinelCPort), pool -> {
+            testRunner.setProperty(redisConnectionPool, RedisUtils.REDIS_MODE, REDIS_MODE_SENTINEL);
+            testRunner.setProperty(redisConnectionPool, RedisUtils.SENTINEL_MASTER, masterName);
+        });
+
+        setupRedisMapCacheClientService();
+
+        executeProcessor();
+    }
+
+    @Test
+    public void testSentinelRedisWithAuthentication() throws InitializationException, IOException {
+        String redisPassword = "t0p_53cr35";
+        String sentinelPassword = "otherPassword";
+
+        RedisContainer redisMasterContainer = setupStandaloneRedis(redisPassword);
+        String masterHost = "127.0.0.1";
+        int masterPort = redisMasterContainer.port;
+        setUpRedisReplica(masterHost, masterPort, redisPassword);
+        setUpRedisReplica(masterHost, masterPort, redisPassword);
+
+        int sentinelAPort = setUpSentinel(masterHost, masterPort, redisPassword, 2, sentinelPassword).port;
+        int sentinelBPort = setUpSentinel(masterHost, masterPort, redisPassword, 2, sentinelPassword).port;
+        int sentinelCPort = setUpSentinel(masterHost, masterPort, redisPassword, 2, sentinelPassword).port;
+
+        setUpRedisConnectionPool(portsToConnectionString(sentinelAPort, sentinelBPort, sentinelCPort), pool -> {
+            testRunner.setProperty(redisConnectionPool, RedisUtils.REDIS_MODE, REDIS_MODE_SENTINEL);
+            testRunner.setProperty(redisConnectionPool, RedisUtils.SENTINEL_MASTER, masterName);
+
+            testRunner.setProperty(redisConnectionPool, RedisUtils.PASSWORD, redisPassword);
+            testRunner.setProperty(redisConnectionPool, RedisUtils.SENTINEL_PASSWORD, sentinelPassword);
+        });
+        setupRedisMapCacheClientService();
+
+        executeProcessor();
+    }
+
+    @AfterEach
+    public void teardown() {
+        if (redisConnectionPool != null) {
+            redisConnectionPool.onDisabled();
+        }
+
+        redisContainers.forEach(RedisContainer::stop);
+    }
+
+    private RedisContainer setupStandaloneRedis(final @Nullable String redisPassword) throws IOException {
+        int redisPort = getAvailablePort();
+
+        RedisContainer redisContainer = new RedisContainer("redis:7.0.12-alpine");
```

Comment:
Recommend setting a static member variable for the image name and version, and reusing it in all locations.
```suggestion
        RedisContainer redisContainer = new RedisContainer(CONTAINER_IMAGE_TAG);
```